_inc/lib/markdown/gfm.php

```diff
@@ -60,7 +60,7 @@ class WPCom_GHF_Markdown_Parser extends MarkdownExtra_Parser {
 	 */
 	public function __construct() {
 		$this->use_code_shortcode  = class_exists( 'SyntaxHighlighter' );
-		$this->preserve_shortcodes = function_exists( 'get_shortcode_regex' );
+		$this->preserve_shortcodes = apply_filters( 'wpcom_markdown_preserve_shortcodes', $this->preserve_shortcodes ) && function_exists( 'get_shortcode_regex' );
```

Comment:
Could you add some inline docs, [like so](https://github.com/lkraav/jetpack/blob/66568dde343d6ff36759322e4621bc5ab9320a92/_inc/lib/markdown/gfm.php#L96)?